File: app/src/main/java/tv/spacedentist/android/SDComponent.java
package tv.spacedentist.android;

import com.google.android.gms.cast.framework.CastContext;

import javax.inject.Singleton;

import dagger.Component;
import tv.spacedentist.android.chromecast.SDChromecastManager;
import tv.spacedentist.android.util.SDLogger;

@Singleton
@Component(modules = SDModule.class)
public interface SDComponent {
    SDLogger getLogger();
    SDChromecastManager getChromecastManager();
    CastContext getCastContext();

    void inject(SDMainActivity mainActivity);
    void inject(SDChromecastManager chromecastManager);
    void inject(SDNotificationManager notificationManager);
}

File: app/src/main/java/tv/spacedentist/android/SDComponent.kt
package tv.spacedentist.android

import com.google.android.gms.cast.framework.CastContext
import dagger.Component
import tv.spacedentist.android.chromecast.SDChromecastManager
import tv.spacedentist.android.util.SDLogger
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(SDModule::class))
interface SDComponent {
    val logger: SDLogger
    val chromecastManager: SDChromecastManager
    val castContext: CastContext

    fun inject(mainActivity: SDMainActivity)
    fun inject(chromecastManager: SDChromecastManager)
    fun inject(notificationManager: SDNotificationManager)
}
